app/models/npc_model.rb

```diff
@@ -0,0 +1,72 @@
+class NpcModel < ApplicationRecord
+  validates :alignment,
+            :ancestry,
+            :armor_class,
+            :background_equipment,
+            :background_extra_languages,
+            :background_name,
+            :background_personality_personality,
+            :background_personality_ideal,
+            :background_personality_bond,
+            :background_personality_flaw,
+            :background_skill_proficiency_one,
+            :background_skill_proficiency_two,
+            :background_tool_proficiency_one,
+            :background_tool_proficiency_two,
+            :character_class,
+            :equipment_extras,
+            :equipment_gold,
+            :gender,
+            :hit_dice,
+            :initiative,
+            :languages,
+            :name,
+            :proficiencies_armor,
+            :proficiencies_weapons,
+            :proficiencies_tools,
+            :size,
+            :speed,
+            :stats_core_stats_stats_modifiers_str_mod,
+            :stats_core_stats_stats_modifiers_dex_mod,
+            :stats_core_stats_stats_modifiers_con_mod,
+            :stats_core_stats_stats_modifiers_int_mod,
+            :stats_core_stats_stats_modifiers_wis_mod,
+            :stats_core_stats_stats_modifiers_cha_mod,
+            :stats_core_stats_stats_scores_str,
+            :stats_core_stats_stats_scores_dex,
+            :stats_core_stats_stats_scores_con,
+            :stats_core_stats_stats_scores_int,
+            :stats_core_stats_stats_scores_wis,
+            :stats_core_stats_stats_scores_cha,
+            :stats_hit_points,
+            :stats_passive_perception,
+            :stats_saving_throws_throws_str,
+            :stats_saving_throws_throws_dex,
+            :stats_saving_throws_throws_con,
+            :stats_saving_throws_throws_int,
+            :stats_saving_throws_throws_wis,
+            :stats_saving_throws_throws_cha,
+            :stats_skills_proficiencies,
+            :stats_skills_acrobatics,
+            :stats_skills_animal_handling,
+            :stats_skills_arcana,
+            :stats_skills_athletics,
+            :stats_skills_deception,
+            :stats_skills_history,
+            :stats_skills_insight,
+            :stats_skills_intimidation,
+            :stats_skills_investigation,
+            :stats_skills_medicine,
+            :stats_skills_nature,
+            :stats_skills_perception,
+            :stats_skills_performance,
+            :stats_skills_persuasion,
+            :stats_skills_religion,
+            :stats_skills_sleight_of_hand,
+            :stats_skills_stealth,
+            :stats_skills_survival,
+            :sub_ancestry,
+            :traits,
+            :vision,
+            :is_saved, presence: true
+end
```

Comment:
Might seem excessive, but there might be even more that's added into here eventually!  There might be a better way to do this upon refactoring, but this is just the start.  I will say all of this makes me wish you could store objects in these models, but alas...